packages/@react-spectrum/provider/src/Provider.tsx

```diff
@@ -189,6 +189,14 @@ export function useProvider(): ProviderContext {
   return useContext(Context);
 }
 
+export function useColorScheme(): ColorScheme {
+  return useProvider().colorScheme;
```

Comment:
I don't know but just guessing that `useProviderProps` expected to work when `context` doesn't exist. I can't think of a scenario where that holds true unless the component isn't wrapped by the provider and in that case I think it is better to fail than default. Let me know if you think otherwise and can change the logic to check for it.
I can do `useProvider()?.colorScheme` instead

Reply:
I think it's fine to throw if it's not within a Provider, otherwise the returned value wouldn't make any sense